### ISSUE TITLE
Python 3.11.0 install doesn’t recognize homebrew Tcl/Tk due to --with-tcltk-libs, --with-tcltk-includes switches being removed from 3.11 

### ISSUE THREAD

**Issue by kjoonlee** [pyenv/pyenv#2499]
### Description
- [X] Platform information: macOS Ventura 13.0, macOS Big Sur 11.7.1
- [X] OS architecture: x86_64, arm64
- [X] pyenv version: 2.3.5
- [X] Python version: 3.11.0
- [X] C Compiler information: Apple clang version 13.0.0 (clang-1300.0.29.30)
- [X] debug trace of the failing command as a gist: https://gist.github.com/kjoonlee/2aac71e965ac6f7db99b880c5d561ddb

I have homebrew python-tk@3.9, python-tk@3.10, and tcl-tk installed.

Python 3.11.0 does not recognize these options:

    configure: WARNING: unrecognized options: --with-tcltk-libs, --with-tcltk-includes

I think it’s an intended change, judging from https://github.com/python/cpython/issues/95957#issuecomment-1214205302

Anyway, so Homebrew tkinter is actually recognized at first:

    ++(/Users/kjoonlee/.pyenv/plugins/python-build/bin/python-build:1558): use_homebrew_tcltk(): can_use_homebrew

But because the switches have been removed, it is reported as missing:

    checking for stdlib extension module _tkinter... missing

So it is not built:

    The necessary bits to build these optional modules were not found:
    _tkinter                                                       
    To find the necessary bits, look in setup.py in detect_modules() for the module's name.

Apologies if anything doesn’t make sense or if it’s unclear. Is there anything I’m doing wrong about the installation, please? Entering a bug just in case.

**Comment by p-i-** [comment]
How are you even getting `pyenv install 3.11` to run?

It doesn't seem to be available:

```bash
> pyenv install 3.11
python-build: definition not found: 3.11

The following versions contain `3.11' in the name:
  3.11.0rc2
  3.11-dev
  miniconda3-4.3.11

See all available versions with `pyenv install --list'.

If the version you need is missing, try upgrading pyenv:

  brew update && brew upgrade pyenv
```
  
Note: I've run ` brew update && brew upgrade pyenv` and still no luck.

Where can I find out the timeline for `pyenv` supporting 3.11?

**Comment by jmarrec** [comment]
Came here today while trying to pyenv install 3.12.2, @kjoonlee's solution didn't work for me, so instead I unset the `TCLTK_xxx` env vars. I have `brew install pkgconf` (a pkgconfig alternative), and `export PKG_CONFIG="$(brew --prefix pkgconf)/bin/pkgconf" so that it's found. And then it worked because python build system picks them correctly.

```shell
$ pkgconf --libs tcl
-L/opt/homebrew/Cellar/tcl-tk/8.6.15/lib -ltcl8.6 -ltclstub8.6
  $ pkgconf --libs tk
-L/opt/homebrew/Cellar/tcl-tk/8.6.15/lib -ltk8.6 -ltkstub8.6 -ltcl8.6 -ltclstub8.6
$ pkgconf --cflags tcl
-I/opt/homebrew/Cellar/tcl-tk/8.6.15/include/tcl-tk
$ pkgconf --cflags tk
-I/opt/homebrew/Cellar/tcl-tk/8.6.15/include/tcl-tk
```

**Comment by oltxae** [comment]
Came here to say that this is still bugged on 3.12 and 3.13 installs. I'm not using a pkgconfig alternative. I get:
`configure: WARNING: unrecognized options: --with-tcltk-libs, --with-tcltk-includes`
on any 3.12 or 3.13 install attempt. considering this was depricated with 3.11 I think the change that fixed 3.11 install tcl flags should be applied here as well

**Comment by oltxae** [comment]
> Workaround: manually setting the flags before running pyenv worked for me:
> 
> On Intel:
> 
> ```
> export TCLTK_LIBS="-L/usr/local/opt/tcl-tk/lib -ltcl8.6 -ltk8.6"
> export TCLTK_CFLAGS="-I/usr/local/opt/tcl-tk/include"
> pyenv install 3.11.0
> ```
> 
> On Apple Silicon:
> 
> ```
> export TCLTK_LIBS="-L/opt/homebrew/opt/tcl-tk/lib -ltcl8.6 -ltk8.6"
> export TCLTK_CFLAGS="-I/opt/homebrew/opt/tcl-tk/include"
> pyenv install 3.11.0
> ```
> 
> `trace.log` and https://github.com/python/cpython/blob/main/Doc/whatsnew/3.11.rst#build-changes gave me the info for this.

I was trying to install 3.11 and found that pyenv wasn't picking up the brew installation of tcl-tk anymore. Found that because tcl-tk was upgraded to version 9, brew installs installs tcl-tk@8 with a different filename now. This breaks the autodetection and the env variables above. Here are new instructions to install 3.11 on MacOS 14 and MacOS 15:

Intel:
```
brew install python-tk@3.11  # this will install tcl-tk@8
export TCLTK_LIBS="-L/usr/local/Cellar/tcl-tk@8/8.6.15/lib -ltcl8.6 -ltk8.6"
export TCLTK_CFLAGS="-I/usr/local/Cellar/tcl-tk@8/8.6.15/include/tcl-tk"
pyenv install 3.11.10
```

Silicon
```
export TCLTK_LIBS="-L/opt/homebrew/Cellar/tcl-tk/8.6.15/lib -ltcl8.6 -ltk8.6"
export TCLTK_CFLAGS="-I/opt/homebrew/Cellar/tcl-tk/8.6.15/include/tcl-tk"
pyenv install 3.11.0
```
hope this helps anyone trying to install 3.11!
